src/libopensc/pkcs15-esteid2025.c

```diff
@@ -50,7 +50,11 @@ sc_pkcs15emu_esteid2025_init(sc_pkcs15_card_t *p15card)
 
 	/* Read document number to be used as serial */
 	sc_format_path("DFDD5007", &tmppath);
-	LOG_TEST_RET(card->ctx, sc_select_file(card, &tmppath, NULL), "SELECT docnr");
+	r = sc_select_file(card, &tmppath, NULL);
+	if (r < 0) {
+		free(buff);
+		LOG_FUNC_RETURN(card->ctx, SC_ERROR_INVALID_CARD);
+	}
```

Comment:
The test in line 59 cannot be true
```
LOG_TEST_RET(card->ctx, r, "read document number failed");
```
could you please remove that line even though it is out of scope of the memory leak?

Reply:
Goo catch! So I think we are good here. Than you! Merging then